ws2801/src/main/java/com/google/android/things/contrib/driver/ws2801/Ws2801.java

```diff
@@ -0,0 +1,51 @@
+package com.google.android.things.contrib.driver.ws2801;
+
+import android.graphics.Color;
+
+import com.google.android.things.pio.PeripheralManagerService;
+import com.google.android.things.pio.SpiDevice;
+
+import java.io.IOException;
+
+public class Ws2801 {
+    private SpiDevice spi_device;
+    private byte[] data;
+    private int nb_leds;
+
+    public Ws2801(int nbleds, String spi_bus_name) throws IOException {
+        nb_leds = nbleds;
+        data = new byte[3*nb_leds];
+        for(int i = 0 ; i < nb_leds ; i++) {
+            data[3*i + 0] = (byte)Color.red(Color.BLUE);
+            data[3*i + 1] = (byte)Color.green(Color.BLUE);
+            data[3*i + 2] = (byte)Color.blue(Color.BLUE);
+        }
+        PeripheralManagerService service = new PeripheralManagerService();
+        spi_device = service.openSpiDevice(spi_bus_name);
+        spi_device.setFrequency(1_000_000);
+        spi_device.setMode(SpiDevice.MODE0);
+        spi_device.setBitsPerWord(8);
+        spi_device.setDelay(1000);
+    }
+
+    public void SetColor(int position, int red, int green, int blue)  throws IOException {
+        SetColor(position, Color.rgb(red, green, blue));
+    }
+
+    public void SetColor(int position, int color) {
```

Comment:
What about exposing a write(android.graphics.Color[]) methods [like](https://github.com/androidthings/contrib-drivers/blob/master/apa102/src/main/java/com/google/android/things/contrib/driver/apa102/Apa102.java#L195) the apa driver instead?

Even if we end up with distinct drivers for each variant of LED strips, it would be nice if they were sharing some convention.